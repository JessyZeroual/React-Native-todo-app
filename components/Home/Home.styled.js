import styled from "styled-components";

export const StyledContainer = styled.View`
  flex: 1;
  background-color: #fff;
`;
