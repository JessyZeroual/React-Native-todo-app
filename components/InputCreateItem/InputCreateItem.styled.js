import styled from "styled-components/native";
import { primaryColor } from "../../constant/style-constants";

export const Form = styled.View`
  display: flex;
  align-items: center;
  flex-direction: row;
  background: #fff;
  height: 40px;
  border-radius: 40px;
  margin: 10px 20px;
  margin-top: -20px;
  padding-left: 20px;
  padding-right: 10px;
`;

export const Input = styled.TextInput`
  flex: 1;
`;

export const ButtonSubmit = styled.TouchableHighlight`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 30px;
  width: 30px;
  background-color: ${primaryColor};
  border-radius: 30px;
`;
