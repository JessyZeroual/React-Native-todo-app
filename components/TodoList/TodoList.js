import React from "react";
import { FlatList, Text } from "react-native";
import { TodoListWrapper } from "./TodoList.styled";

const TodoList = ({ todos }) => {
  const renderItem = ({ item }) => <Text>{item.title}</Text>;
  return (
    <TodoListWrapper>
      <FlatList
        data={todos}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
      />
    </TodoListWrapper>
  );
};

export default TodoList;
