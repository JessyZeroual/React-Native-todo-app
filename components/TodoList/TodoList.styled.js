import styled from "styled-components/native";
import { primaryColor } from "../../constant/style-constants";

export const TodoListWrapper = styled.View`
  background: #ebebef;
  padding: 0px 10px;
  width: 100%;
`;
