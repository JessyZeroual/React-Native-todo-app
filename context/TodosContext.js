import React from "react";

const TodosContext = React.createContext({});

export const TodosContextProvider = TodosContext.Provider;
export const TodosContextConsumer = TodosContext.Consumer;
export default TodosContext;
