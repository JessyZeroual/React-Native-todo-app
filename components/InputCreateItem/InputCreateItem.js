import React, { useState, useContext } from "react";
import TodosContext from "../../context/TodosContext";
import Icon from "react-native-vector-icons/MaterialIcons";
import {
  InputCreateItemWrapper,
  Shape,
  Form,
  Input,
  ButtonSubmit,
} from "./InputCreateItem.styled";

const InputCreateItem = () => {
  const [text, setText] = useState("");
  const todosContext = useContext(TodosContext);
  const { _postTodo } = todosContext.dispatch;
  return (
    <InputCreateItemWrapper>
      <Shape />
      <Form style={shadow}>
        <Input
          onChangeText={(text) => setText(text)}
          placeholder="Add item"
          placeholderTextColor="grey"
          value={text}
          onSubmitEditing={() => _postTodo(text).then(setText(""))}
        />

        <ButtonSubmit onPress={() => _postTodo(text).then(setText(""))}>
          <Icon name="add" color="#fff" size={20}></Icon>
        </ButtonSubmit>
      </Form>
    </InputCreateItemWrapper>
  );
};

const shadow = {
  shadowColor: "#000",
  shadowOffset: {
    width: 0,
    height: 1,
  },
  shadowOpacity: 0.2,
  shadowRadius: 1.41,

  elevation: 2,
};

export default InputCreateItem;
