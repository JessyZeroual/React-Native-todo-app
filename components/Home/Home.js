import React, { useContext } from "react";
import { ActivityIndicator, View } from "react-native";

import TodosContext from "../../context/TodosContext";

import { Container } from "./Home.styled";
import { ViewCenter } from "../../style/common.styled";

import TodoList from "../TodoList/TodoList";
import InputCreateItem from "../InputCreateItem/InputCreateItem";

const Home = () => {
  const todosContext = useContext(TodosContext);
  const { isLoading } = todosContext.state;

  return (
    <Container>
      {isLoading ? (
        <ViewCenter>
          <ActivityIndicator size="large" />
        </ViewCenter>
      ) : (
        <View>
          <InputCreateItem />
          <TodoList />
        </View>
      )}
    </Container>
  );
};

export default Home;
