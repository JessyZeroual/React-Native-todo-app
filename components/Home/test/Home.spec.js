import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { ActivityIndicator } from "react-native";
import { TodosContextProvider } from "../../../context/TodosContext";
import Home from "../Home";
import InputCreateItem from "../../InputCreateItem/InputCreateItem";
import TodoList from "../../TodoList/TodoList";

jest.mock("../../InputCreateItem/InputCreateItem", () => () =>
  "InputCreateItem"
);
jest.mock("../../TodoList/TodoList", () => () => "TodoList");

describe("Home", () => {
  let renderer;

  describe("When Home is mounted", () => {
    beforeEach(() => {
      act(() => {
        renderer = TestRenderer.create(
          <TodosContextProvider
            value={{
              state: { isLoading: false },
            }}
          >
            <Home />
          </TodosContextProvider>
        );
      });
    });

    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("display component InputCreateItem", () => {
      expect(renderer.root.findByType(InputCreateItem)).toBeTruthy();
    });

    it("display components TodoList", () => {
      expect(renderer.root.findByType(TodoList)).toBeTruthy();
    });
  });

  describe("When Home is unmounted", () => {
    beforeEach(() => {
      act(() => {
        renderer = TestRenderer.create(
          <TodosContextProvider
            value={{
              state: { isLoading: true },
            }}
          >
            <Home />
          </TodosContextProvider>
        );
      });
    });

    it("display ActivityIndicator ", () => {
      expect(renderer.root.findByType(ActivityIndicator)).toBeTruthy();
    });
  });
});
