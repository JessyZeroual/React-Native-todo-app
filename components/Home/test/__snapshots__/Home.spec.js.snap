// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home When Home is mounted renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View>
    InputCreateItem
    TodoList
  </View>
</View>
`;
